PowerEditor/src/ScitillaComponent/ScintillaEditView.cpp

```diff
@@ -2414,9 +2414,9 @@ void ScintillaEditView::convertSelectedTextTo(bool Case)
 		for (int i = 0 ; i < nbChar ; ++i)
 		{
 			if (Case == UPPERCASE)
-				selectedStrW[i] = (WCHAR)::CharUpperW((LPWSTR)selectedStrW[i]);
+				selectedStrW[i] = (WCHAR)(UINT_PTR)::CharUpperW((LPWSTR)selectedStrW[i]);
 			else
-				selectedStrW[i] = (WCHAR)::CharLowerW((LPWSTR)selectedStrW[i]);
+				selectedStrW[i] = (WCHAR)(UINT_PTR)::CharLowerW((LPWSTR)selectedStrW[i]);

```

Comment:
I don't really agree that these are better. Sure, [`CharUpper`](https://msdn.microsoft.com/en-us/library/windows/desktop/ms647474)/[`CharLower`](https://msdn.microsoft.com/en-us/library/windows/desktop/ms647467) are really terrible APIs, but adding *more* C-style casting just serves to make it *more* confusing. 

[I proposed a patch that included a "fix" for this, a while ago](https://sourceforge.net/p/notepad-plus/patches/662/), the patch was far too big to have been accepted, but I think that what I did for `CharUpper`/`CharLower` was much clearer.

Essentially what I did was first, extract the low-order word:
```
PWSTR fake_lpsz = PWSTR( LOWORD( destStr[ j ] ) );
```
...then pass `fake_lpsz` to `CharUpperW`:

```
PCWSTR const fake_lpsz_result = ::CharUpperW( fake_lpsz );
```
....and finally, assigned the value to `destStr`:

```
destStr[ j ] = LOWORD( fake_lpsz_result );
```

That has the benefit of being explicit - you're saying "give me **that** *specific word*" instead of "shove the whole damned thing in a `WCHAR`, and don't complain if it doesn't fit!" - thus it's more obviously not [a crazy bug](http://www.viva64.com/en/b/0083/). *Seriously, it looks insane before __and__ after reading the documentation for `CharUpper`*.


A screencap, for your convenience:
![npp_charupper](https://cloud.githubusercontent.com/assets/2142308/7720073/0ef51c40-fe97-11e4-8965-3068e29582bd.PNG)